pkg/auth/handler/webapp/authflowv2/settings_identity_email.go

```diff
@@ -0,0 +1,132 @@
+package authflowv2
+
+import (
+	"net/http"
+	"sort"
+
+	handlerwebapp "github.com/authgear/authgear-server/pkg/auth/handler/webapp"
+	"github.com/authgear/authgear-server/pkg/auth/handler/webapp/viewmodels"
+	"github.com/authgear/authgear-server/pkg/auth/webapp"
+	"github.com/authgear/authgear-server/pkg/lib/session"
+	"github.com/authgear/authgear-server/pkg/util/httproute"
+	"github.com/authgear/authgear-server/pkg/util/template"
+)
+
+var TemplateWebSettingsIdentityEmailHTML = template.RegisterHTML(
+	"web/authflowv2/settings_identity_email.html",
+	handlerwebapp.SettingsCompenents...,
+)
+
+func ConfigureSettingsIdentityEmailRoute(route httproute.Route) httproute.Route {
+	return route.
+		WithMethods("OPTIONS", "POST", "GET").
+		WithPathPattern("/settings/identity/email")
+}
+
+type CheckedEmail struct {
+	Email      string
+	Verified   bool
+	Verifiable bool
+	UpdatedAt  int64
+}
+
+type AuthflowV2SettingsIdentityEmailHandler struct {
+	ControllerFactory        handlerwebapp.ControllerFactory
+	BaseViewModel            *viewmodels.BaseViewModeler
+	SettingsProfileViewModel *viewmodels.SettingsProfileViewModeler
+	Identities               viewmodels.SettingsProfileIdentityService
+	Renderer                 handlerwebapp.Renderer
+	Verification             handlerwebapp.SettingsVerificationService
+}
+
+func (h *AuthflowV2SettingsIdentityEmailHandler) GetData(r *http.Request, rw http.ResponseWriter) (map[string]interface{}, error) {
+	data := map[string]interface{}{}
+
+	baseViewModel := h.BaseViewModel.ViewModel(r, rw)
+	viewmodels.Embed(data, baseViewModel)
+
+	userID := session.GetUserID(r.Context())
+
+	settingsProfileViewModel, err := h.SettingsProfileViewModel.ViewModel(*userID)
+	if err != nil {
+		return nil, err
+	}
+	viewmodels.Embed(data, *settingsProfileViewModel)
+
+	identities, err := h.Identities.ListByUser(*userID)
+	if err != nil {
+		return nil, err
+	}
+
+	verifications, err := h.Verification.GetVerificationStatuses(identities)
+	if err != nil {
+		return nil, err
+	}
+
+	emails := []CheckedEmail{}
+	for _, id := range identities {
+		claims, ok := verifications[id.ID]
+		if !ok || len(claims) == 0 {
+			continue
+		}
+
+		claim := claims[0]
```

Comment:
What if there are more than 1 claim. How would we handle it?

Reply:
I wouldn't know :doglaugh: I just reuse the logic I see in the authflowv1. Maybe it's stored in reverse chronological order?